src/battle-dex-data.ts

```diff
@@ -1370,7 +1370,7 @@ class Species implements Effect {
 		this.battleOnly = data.battleOnly || (this.isGigantamax ? this.baseSpecies : undefined);
 		this.isNonstandard = data.isNonstandard || null;
 		this.unreleasedHidden = data.unreleasedHidden || false;
-		this.changesFrom = data.changesFrom || undefined;
+		this.changesFrom = data.changesFrom || (this.isGigantamax ? this.baseSpecies : undefined);
```

Comment:
The `battleOnly` check should cover it; it 100% does not need to be repeated here.

Reply:
It doesn't.

Reply:
Then it should be fixed in the implementation, not here.

Reply:
Like you didn't update the documentation in `FORMES.md`, nor the jsdoc in `dex-data.ts`. Those are things you _must_ do if you want to change what a property means.

But at some point you should probably just realize that you don't actually want to change what the property means, but just how it's used.